include/lsst/afw/detection/Psf.h

```diff
@@ -29,11 +29,56 @@
 #include <memory>
 
 #include "lsst/daf/base.h"
+#include "lsst/utils/CacheFwd.h"
+#include "lsst/utils/hashCombine.h"
 #include "lsst/afw/geom/ellipses/Quadrupole.h"
 #include "lsst/afw/math/Kernel.h"
 #include "lsst/afw/image/Color.h"
 #include "lsst/afw/table/io/Persistable.h"
 
+namespace lsst::afw::detection::detail {
+
+/** Key for caching PSFs with lsst::utils::Cache
+ *
+ * We cache PSFs by their x,y position. Although there are placeholders
+ * in the `Psf` class and here for `image::Color`, these are not used
+ * in the cache because `image::Color` is not currently well-defined
+ * or used.
+ */
+struct PsfCacheKey {
+    geom::Point2D position;
+    image::Color color;
+
+    PsfCacheKey(geom::Point2D const& position_, image::Color color_=image::Color())
+      : position(position_), color(color_) {}
+
+    bool operator==(PsfCacheKey const& other) const {
+        return position == other.position;  // Currently don't care about color
+    }
+
+    friend std::ostream & operator<<(std::ostream &os, PsfCacheKey const &key) {
+        return os << key.position;
+    }
+};
```

Comment:
You'll need a forward declaration of this class in the header for the typedef later, but the rest can also go in `Psf.cc`.